### PR TITLE
add annotations for tuning headless proxies

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

**Review by nluaces — APPROVED** [review]

**Comment by grs** [comment]
resolves remaining issue in #896

**Review by fgiorgetti — APPROVED** [review]
LGTM. One minor comment only, probably can be discussed separately.

File: cmd/service-controller/definition_monitor.go [hunk 1]
```diff
@@ -165,6 +166,9 @@ func updateAnnotatedServiceDefinition(actual *types.ServiceInterface, desired *t
 	if actual.Protocol != desired.Protocol || !reflect.DeepEqual(actual.Ports, desired.Ports) {
 		return true
 	}
+	if actual.Headless != desired.Headless {
+		return true
```
[the remark this answers is not in the document]
Comment (grs) [reply cmd/service-controller/definition_monitor.go:170]: If the headless option changes (as is the case if protocol or ports change), then the targets are not additive. I.e. it is considered a reset of the definition.

The need for the additive logic for targets comes from the case where you have e.g. two Deployments that both specify the same address. The headless mode would not work for such cases.

I think therefore that this is correct. Does that make sense?

[the remark this answers is not in the document]
Comment (fgiorgetti) [reply cmd/service-controller/definition_monitor.go:170]: Sounds right to me. Thanks!

**Review by ajssmith — APPROVED** [review]
lgtm, reviewed, ran tests locally and against cluster

**Pull request merged**